# World Bank's International Debt

In this notebook I will be analysing international debt data collected by the World Bank. This contains data covering the debt owed by countries (USD) across several categories.

Covering questions including:

<li>What is the total amount of debt owed by each country in the dataset?
<li>What categories of debt are the most prelevant amongst the countries?
<li>What countries are under the most debt?

In [97]:
import psycopg2 as pg2
import sqlalchemy as sa

In [84]:
conn  = pg2.connect(database='International_Debt',user='postgres',password='kieran')
cur = conn.cursor()
engine = sa.create_engine('postgresql://postgres:kieran@localhost:5438/postgres')

## At First Glance
<p> Of the first 10 rows in the dataset we see the different debts owed by <em>Afganistan</em> in their respective types as shown by the indicator codes and names. From this we can not yet gather the amount of countries in the dataset or debt values from these different countries to compare to.</p>

<p>To get a better idea of the dataset we need to know the amount of unique countries in the list which would enable for more statistical analysis.


In [85]:
%load_ext sql
%sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql
 * postgresql://postgres:***@localhost/International_Debt


In [104]:
%%sql 
SELECT
    country_name,
    country_code,
    indicator_name,
    indicator_code,
    ROUND(debt,0) as debt
FROM international_debt
LIMIT 10

 * postgresql://postgres:***@localhost/International_Debt
10 rows affected.


[('Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('72894454')),
 ('Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('53239440')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('61739337')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('49114729')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('39903620')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (AMT, current US$)', 'DT.AMT.MLAT.CD', Decimal('39107845')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (DIS, current US$)', 'DT.DIS.MLAT.CD', Decimal('23779724')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (INT, current US$)', 'DT.INT.MLAT.CD', Decimal('13335820')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (AMT, current US$)', 'DT.AMT.OFFT.CD', Decimal('100847182')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (DIS, current US$)', 'DT.DIS.OFFT.CD', Decimal('72894454'))]

In [96]:
%%sql
SELECT COUNT(DISTINCT(country_code)) as total_distinct_countries
FROM international_debt

 * postgresql://postgres:***@localhost/International_Debt
1 rows affected.


[(124,)]

## Unique debt indicators

<p>We can see there are 124 countries present on the table. As we saw in the earlier sections, there are columns called <code>indicator_code</code> and <code>indicator_name</code> that show the codes of the specific debts and the named version of them respectively</p>

In [98]:
%%sql
SELECT DISTINCT(indicator_code) AS distinct_debt_indicators,
       indicator_name
FROM international_debt
ORDER BY distinct_debt_indicators;

 * postgresql://postgres:***@localhost/International_Debt
25 rows affected.


[('DT.AMT.BLAT.CD', 'PPG, bilateral (AMT, current US$)'),
 ('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)'),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)'),
 ('DT.AMT.MLAT.CD', 'PPG, multilateral (AMT, current US$)'),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)'),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)'),
 ('DT.AMT.PCBK.CD', 'PPG, commercial banks (AMT, current US$)'),
 ('DT.AMT.PROP.CD', 'PPG, other private creditors (AMT, current US$)'),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)'),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)'),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)'),
 ('DT.DIS.MLAT.CD', 'PPG, multilateral (DIS, current US$)'),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)'),
 ('DT.DIS.PCBK.CD', 'PPG, commercial banks (DIS, current US$)'),
 ('DT.DIS.PROP.CD', 'PPG, other private creditors (DIS, current US$)'),
 ('DT.DIS.PRVT.CD', 'PPG, private creditors (DIS, current US$)'),
 ('DT.INT.BLAT.CD', 'PPG, bilateral (INT, current US$)'),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)'),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)'),
 ('DT.INT.MLAT.CD', 'PPG, multilateral (INT, current US$)'),
 ('DT.INT.OFFT.CD', 'PPG, official creditors (INT, current US$)'),
 ('DT.INT.PBND.CD', 'PPG, bonds (INT, current US$)'),
 ('DT.INT.PCBK.CD', 'PPG, commercial banks (INT, current US$)'),
 ('DT.INT.PROP.CD', 'PPG, other private creditors (INT, current US$)'),
 ('DT.INT.PRVT.CD', 'PPG, private creditors (INT, current US$)')]

## Overall Debt

The financial debt of a country represents its economics state individually. However to broaden the scope if we needed an overall understanding of the world's debt it could give us a new perspective of the world economy. 

The total debt in the world shown below is in millions, displaying the sheer amount of debt our countries overall have right now.

In [105]:
%%sql 
SELECT ROUND(SUM(debt)/1000000,2) AS total_debt
FROM international_debt

 * postgresql://postgres:***@localhost/International_Debt
1 rows affected.


[(Decimal('3079734.49'),)]

## Countries with the highest debt

Now that we have a general idea of the amount of debt owed by countries along with respective codes. We can find the countries with the highest amount of debt giving us more understanding of these countries' socio-economic statuses and if these match up with our predetermined assumptions. 

In [106]:
%%sql
SELECT country_name, ROUND(SUM(debt),2) as total_debt
FROM international_debt
GROUP BY international_debt.country_name
ORDER BY total_debt DESC
LIMIT 3;

 * postgresql://postgres:***@localhost/International_Debt
3 rows affected.


[('China', Decimal('285793494734.20')),
 ('Brazil', Decimal('280623966140.80')),
 ('South Asia', Decimal('247608723990.60'))]

## Average debt across indicators

As one of the largest economies <em>China</em> seemed plausible for having the largest amount of debt followed by Brazil and South Asia which had large increases in <a href="https://thehill.com/opinion/finance/578527-a-brazilian-budget-cautionary-tale-for-the-united-states">public spending</a> and <a href="https://www.worldbank.org/en/news/press-release/2021/06/24/south-asia-must-reform-debt-accumulating-state-owned-banks-and-enterprises#:~:text=%E2%80%9CThe%20COVID%2D19%20pandemic%20has,Vice%20President%20for%20South%20Asia">hidden debt</a> respectively.

Now we have a brief overview of the dataset, we can dig further into the average amount of debt owed under each indicator showing us how well distributed the debt is under each

In [92]:
%%sql 
SELECT 
    indicator_code AS debt_indicator,
    indicator_name,
    ROUND(AVG(debt)) AS average_debt
FROM international_debt
GROUP BY debt_indicator, indicator_name
ORDER BY average_debt DESC
LIMIT 10;

 * postgresql://postgres:***@localhost/International_Debt
10 rows affected.


[('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('5904868401')),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', Decimal('5161194334')),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)', Decimal('2152041217')),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('1958983453')),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)', Decimal('1803694102')),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)', Decimal('1644024068')),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)', Decimal('1223139290')),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', Decimal('1220410844')),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', Decimal('1191187963')),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)', Decimal('1082623948'))]

## Highest debt codes

We observe that the indicator <code>DT.AMT.DLXF.CD</code> contains the highest average debt. This entails the repayment of long term debts which are used for immediate capital.

It is also worth noting that the gap between the top two debt categorgies and the rest is very large, indicating the first two debt categories are the most severe for countries in terms of debt.

Taking the first indicator as an example, we can investigate which country owes the highest amount of debt in this category. Looking at the can enable us to see this country's economic situation more clearly.

In [93]:
%%sql
SELECT
    country_name,
    indicator_name
FROM international_debt
WHERE debt = (SELECT MAX(debt)
              FROM international_debt
              WHERE indicator_code = 'DT.AMT.DLXF.CD');

 * postgresql://postgres:***@localhost/International_Debt
1 rows affected.


[('China', 'Principal repayments on external debt, long-term (AMT, current US$)')]

## The most common debt indicator

China has the highest amount of debt in the long term debt categeory as shown above. 

We observed that the long-term debt is the highest category in terms of average debt. But what about the most common type of debt to these said countries?

In [94]:
%%sql
SELECT indicator_code, COUNT(indicator_code) as indicator_count
FROM international_debt
GROUP BY indicator_code
ORDER BY indicator_count DESC, indicator_code DESC

 * postgresql://postgres:***@localhost/International_Debt
25 rows affected.


[('DT.INT.OFFT.CD', 124),
 ('DT.INT.MLAT.CD', 124),
 ('DT.INT.DLXF.CD', 124),
 ('DT.AMT.OFFT.CD', 124),
 ('DT.AMT.MLAT.CD', 124),
 ('DT.AMT.DLXF.CD', 124),
 ('DT.DIS.DLXF.CD', 123),
 ('DT.INT.BLAT.CD', 122),
 ('DT.DIS.OFFT.CD', 122),
 ('DT.AMT.BLAT.CD', 122),
 ('DT.DIS.MLAT.CD', 120),
 ('DT.DIS.BLAT.CD', 113),
 ('DT.INT.PRVT.CD', 98),
 ('DT.AMT.PRVT.CD', 98),
 ('DT.INT.PCBK.CD', 84),
 ('DT.AMT.PCBK.CD', 84),
 ('DT.INT.DPNG.CD', 79),
 ('DT.AMT.DPNG.CD', 79),
 ('DT.INT.PBND.CD', 69),
 ('DT.AMT.PBND.CD', 69),
 ('DT.INT.PROP.CD', 54),
 ('DT.AMT.PROP.CD', 54),
 ('DT.DIS.PRVT.CD', 53),
 ('DT.DIS.PCBK.CD', 51),
 ('DT.DIS.PROP.CD', 19)]

## Conclusion

In this exploratory analysis we looked at the debt owned by countries across the world. Extracting a few statistics from the dataset to draw insights on the top debt holding countries and their respective categories